### PR TITLE
also check function literals, fixes #19

### LINKED ISSUE #19: Does not work in table-driven tests
One standard pattern in go is to used table driven tests, see https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

This linter does not work if  `os.Setenv` is used in these tests.

Example:
```
func TestLintShouldFailButDoesnt(t *testing.T) {
	tests := map[string]struct {
		fooValue string
	}{
		"does something": {fooValue: "bar"},
	}

	for name, tc := range tests {
		t.Run(name, func(t *testing.T) {
			os.Setenv("foo", tc.fooValue)
		})
	}
}
```

I created this on my repo with the github-action version of the golang-ci linter:
https://github.com/vankleefjim/tenv-bug/commit/10f12e411fc285d5abe19fcb9ad63d9ef0f139be with successful CI run
https://github.com/vankleefjim/tenv-bug/commit/f01c6891a6020c613e9cbccf5df46dc5dd71e188 with failing CI run, but only in the test that directly runs os.Setenv without a table-driven test.

### PR DESCRIPTION
**Pull request by vankleefjim**

### CODE REVIEW

File: testdata/src/a/a_test.go [hunk 1]
```diff
@@ -55,3 +55,15 @@ func FuzzF(f *testing.F) {
 		_ = err
 	}
 }
+
+func TestFunctionLiteral(t *testing.T) {
+	testsetup()
+	t.Run("test", func(t *testing.T) {
+		os.Setenv("a", "b")        // want "os\\.Setenv\\(\\) can be replaced by `t\\.Setenv\\(\\)` in function literal"
```
Comment (sivchari) [inline testdata/src/a/a_test.go]: I think this report message is ambiguous. `function literal` is compiler(token/parser etc..) keyword. So, The someone uses a `tenv` may not know the word. Therefore, I wanna improve the message (e.g. Replace to `function literal` to `subtest name`).

**Review by sivchari — COMMENTED** [review]
Good catch ! Thanks @vankleefjim . Just nits about a report message.

File: testdata/src/a/a_test.go (hunk 1 above)
Comment (vankleefjim) [reply testdata/src/a/a_test.go]: Thank you for your feedback. I don't know how to find the subtest name without making this analyzer a lot more complicated, do you think just using `anonymous function` would make it clear enough? Especially as the exact file+line is already given to the user

Comment (sivchari) [reply testdata/src/a/a_test.go]: Looks good. Actually, a subtest isn't given function name, so I think `anonymous function` is correct.

**Review by sivchari — APPROVED** [review]
LGTM

**Comment by sivchari** [comment]
Please fix lint error.

**Comment by sivchari** [comment]
Nice, I'll merge later 👍 .

**Pull request merged**